src/libexpr-c/nix_api_expr.cc

```diff
@@ -106,7 +107,21 @@ EvalState * nix_state_create(nix_c_context * context, const char ** lookupPath_c
             for (size_t i = 0; lookupPath_c[i] != nullptr; i++)
                 lookupPath.push_back(lookupPath_c[i]);
 
-        return new EvalState{nix::EvalState(nix::LookupPath::parse(lookupPath), store->ptr)};
+        void * p = ::operator new(
+            sizeof(EvalState),
+            static_cast<std::align_val_t>(alignof(EvalState)));
```

Comment:
This is causing problems for 24.05.
Might a simpler solution be feasible?

https://github.com/NixOS/nixpkgs/pull/324408/checks?check_run_id=27026504754